src/js/common/schemaform/save-in-progress/helpers.js

```diff
@@ -0,0 +1,65 @@
+import {
+  createPageList,
+  createFormPageList
+} from '../helpers';
+import RoutedSavablePage from './RoutedSavablePage';
+import RoutedSavableReviewPage from './RoutedSavableReviewPage';
+import FormSaved from './FormSaved';
+import SaveInProgressErrorPage from './SaveInProgressErrorPage';
+
+export function addSaveInProgressRoutes(config, routes) {
+  const protectedRoutes = ['introduction', 'review-and-submit', 'confirmation', '*'];
+  const formPages = createFormPageList(config);
+  const pageList = createPageList(config, formPages);
+
+  const newRoutes = routes;
+
+  newRoutes.forEach((route) => {
+    const indexOfRoute = newRoutes.indexOf(route);
```

Comment:
`forEach` passes the current value `index` as the second argument to this callback.

Reply:
ah, good catch thanks